src/forms/DropHandler.js

```diff
@@ -0,0 +1,82 @@
+const DropHelper = {
+
+  // Putting this function inside the DropHandler would cause an
+  // invariant violation on ReactDnD (see: http://gaearon.github.io/react-dnd/docs-drop-target.html)
+  showDropCandidate(draggedItem, container, targetPosition, component) {
+
+    let currentFields = container.state.savedFields.slice();
+
+    // If item was present on the list
+    if (draggedItem.onList) {
+
+      // First we make a copy removing the dragged element
+      let fieldsCopy = currentFields.slice();
+      fieldsCopy.splice(draggedItem.position, 1);
+
+      let fieldsBefore = fieldsCopy.slice(0, targetPosition);
+      let fieldsAfter = fieldsCopy.slice(targetPosition);
+      currentFields = fieldsBefore.concat(draggedItem.field).concat(fieldsAfter);
```

Comment:
You're right!, I tend to avoid `splice` as it's a mutating method but I'm already working on a copy here. Now that I see it, `currentFields` is already a copy... :P